pyjelly/integrations/rdflib/parse.py

```diff
@@ -82,83 +84,47 @@ class RDFLibTriplesAdapter(RDFLibAdapter):
         triples and namespaces and can get flushed between frames.
     """
 
-    graph: Graph
-
-    def __init__(self, options: ParserOptions, store: Store | str = "default") -> None:
+    def __init__(self, options: ParserOptions, graph: Graph) -> None:
         super().__init__(options=options)
-        self.graph = Graph(store=store)
+        self.graph = graph
 
     @override
     def triple(self, terms: Iterable[Any]) -> Any:
-        self.graph.add(terms)  # type: ignore[arg-type]
+        self.graph.add(tuple(terms))
 
     @override
     def namespace_declaration(self, name: str, iri: str) -> None:
         self.graph.bind(name, self.iri(iri))
 
-    def frame(self) -> Graph | None:
+    def frame(self) -> None:
         """
         Finalize one frame in triples stream.
 
         Returns:
-            Graph | None: graph if logical type is GRAPHS and starts a new graph
-                if flat/unspecified, does not wrap each frame into a separate graph
-                and just leaves everything in one graph
-                TODO: this logic breaks conformance tests bc it keeps adding
+           None
+           TODO: this logic breaks conformance tests bc it keeps adding
                 triples to one graph despite frames division
 
         """
-        if self.options.stream_types.logical_type == jelly.LOGICAL_STREAM_TYPE_GRAPHS:
-            this_graph = self.graph
-            self.graph = Graph(store=self.graph.store)
-            return this_graph
-        if self.options.stream_types.logical_type in (
-            jelly.LOGICAL_STREAM_TYPE_UNSPECIFIED,
-            jelly.LOGICAL_STREAM_TYPE_FLAT_TRIPLES,
-        ):
-            return None
-        return _adapter_missing(
-            "interpreting frames",
-            stream_types=self.options.stream_types,
-        )
 
 
 class RDFLibQuadsBaseAdapter(RDFLibAdapter):
     def __init__(
         self,
         options: ParserOptions,
-        store: Store | str,
+        dataset: Dataset,
     ) -> None:
         super().__init__(options=options)
-        self.store = store
-        self.dataset = self.new_dataset()
-
-    def new_dataset(self) -> Dataset:
-        return Dataset(store=self.store, default_union=True)
+        self.dataset = dataset
 
     @override
-    def frame(self) -> Dataset | None:
+    def frame(self) -> None:
         """
         Finalize one frame in quads stream.
 
-        Returns:
-            Dataset | None: returns frame content as dataset if
-                logical type = DATASETS and starts a new dataset,
-                otherwise, do nothing
+        TODO: what should happen here?
```

Comment:
We should have a parameter telling us whether this is flat or grouped parsing.

If grouped, we construct a new dataset per frame.
If flat, we use the same one.

To do this, you will need to replace the `dataset` parameter with a dataset factory.